src/clients/gists.js

```diff
@@ -81,16 +87,78 @@ function canUpdateGist(user) {
   const uri = document.createElement('a');
   uri.setAttribute('href', '/');
   uri.search = `gist=${gistData.id}`;
 
-  const description = `${gistData.description} Click to import: ${uri.href}`;
+  const description =
+      `${gistData.description} Click to import: ${uri.href} 
+      \n Shareable URL: ${shareableUrl}`;
   const response = await performWithRetries(() => gist.update({description}));
   return response.data;
 }
 
+async function createShareableURL(rawGitUrl) {
+  const uri =
+      `https://api-ssl.bitly.com/v3/shorten?access_token=ceaf89abb8773e91ba80a54a7c0fa7d683749740&longUrl=${rawGitUrl}`;
+  const response = await performWithRetries(() => fetch(uri));
+  const data = await response.json();
+  return data.data.url;
+}
+
+function createRawGitURL(gistData) {
+  const rawUrl = gistData.files['preview.html'].raw_url;
+  const rawGitUrl = `https://cdn.rawgit.com${rawUrl.slice(34, rawUrl.length)}`;
+  return rawGitUrl;
+}
+
+function createPreviewHtml(project) {
+  const previewHtml = document.createElement('html');
+  const indexHtml = trim(project.sources.html);
+
+  previewHtml.innerHTML = indexHtml;
+
+  const sheet = document.createElement('style');
+  sheet.innerHTML = trim(project.sources.css);
+  previewHtml.appendChild(sheet);
+
+  const script = document.createElement('script');
+  script.innerHTML = trim(project.sources.javascript);
+  previewHtml.appendChild(script);
+
+  if (project.enabledLibraries.includes('jquery')) {
+    const jqueryScript = document.createElement('script');
+    jqueryScript.setAttribute('src', 'https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js');
+    previewHtml.getElementsByTagName('head')[0].appendChild(jqueryScript);
+  }
+  if (project.enabledLibraries.includes('lodash')) {
+    const lodashScript = document.createElement('script');
+    lodashScript.setAttribute('src', 'https://cdnjs.cloudflare.com/ajax/libs/lodash.js/4.17.4/lodash.min.js');
+    previewHtml.getElementsByTagName('head')[0].appendChild(lodashScript);
+  }
+  if (project.enabledLibraries.includes('mustache')) {
+    const mustacheScript = document.createElement('script');
+    mustacheScript.setAttribute('src', 'https://cdnjs.cloudflare.com/ajax/libs/mustache.js/2.3.0/mustache.min.js');
+    previewHtml.getElementsByTagName('head')[0].appendChild(mustacheScript);
+  }
+  if (project.enabledLibraries.includes('bootstrap')) {
+    const bootstrapLink = document.createElement('link');
+    bootstrapLink.setAttribute('href', 'https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css');
+    bootstrapLink.setAttribute('rel', 'stylesheet');
+    previewHtml.getElementsByTagName('head')[0].appendChild(bootstrapLink);
+
+    const bootstrapScript = document.createElement('script');
+    bootstrapScript.setAttribute('src', 'https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js');
+    previewHtml.getElementsByTagName('head')[0].appendChild(bootstrapScript);
+  }
```

Comment:
Yeah that was the thought. Definitely an ugly solution.  Anyway, I think it makes sense to stop working on this, but it was fun learning more about this process!